test/pass/transpose_sinking_test.cpp

```diff
@@ -0,0 +1,337 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <algorithm>
+#include <cstdio>
+#include <iostream>
+#include <list>
+#include <memory>
+
+#include "gtest/gtest.h"
+
+#include "ngraph/ngraph.hpp"
+#include "ngraph/opsets/opset3.hpp"
+#include "ngraph/pass/manager.hpp"
+#include "ngraph/pass/pass.hpp"
+#include "ngraph/pattern/op/label.hpp"
+#include "ngraph/util.hpp"
+
+#include "logging/tf_graph_writer.h"
+#include "ngraph_bridge/ngraph_utils.h"
+#include "ngraph_bridge/pass/transpose_sinking.h"
+#include "test/opexecuter.h"
+#include "test/test_utilities.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+namespace ngraph_bridge {
+namespace testing {
+
+TEST(TransposeSinking, PassProperty) {
+  auto pass = std::make_shared<pass::TransposeSinking>();
+  ASSERT_TRUE(
+      pass->get_property(ngraph::pass::PassProperty::REQUIRE_STATIC_SHAPE));
+  ASSERT_FALSE(
+      pass->get_property(ngraph::pass::PassProperty::CHANGE_DYNAMIC_STATE));
+}
+
+TEST(TransposeSinking, EdgeSplitting) {
+  // checks if Transpose is pushed through ng::opset3::Abs, but stopped by
```

Comment:
Wondering if this test needs to be changed so the output shape is NHWC instead of NCHW

Reply:
Should it matter one way or another?

Reply:
Not sure, may be it does not since it is a unit test and we aren't comparing with TF.